### PR TITLE
ALC now has units for the x axis and point data

### LINKED ISSUE #31482: Include Units With Sample Logs Muon Analysis And ALC
Within the NeXus files some logged values have associated units which are stored as an attribute. This should be seen when creating a plot against the log. For example, in the ALC interface if doing a plot of the log of the HIFI main field, the plot should say something like "Field_Main (G)" with the units in brackets instead of just "Field_Main"

### PR DESCRIPTION
**Pull request by AnthonyLim23**
**Description of work.**
The ALC interface was not showing log units in the plots. This PR fixes that. Once the units were added it was noticed that the plots were normalised to bin width, so I also changed them to have no normalisation. 

**To test:**
Open ALC
Load some data (e.g. EMU 20889-20900)
Select a log value that has units (some do not) (e.g. Field_Danfysik)
Click load
Check the plot has the correct axis labels and units

Go to the baseline modelling (click the button on the bottom left)
Add a fit function
Add a fit region
Do a fit
Check the plot has the correct axis labels and units
Also check the plot on the corrected data tab

Go to the peak fitting (button on bottom left)
Add a peak with some sensible values
Click plot guess 
Check the plot has the correct axis labels and units
Click fit
Check the plot has the correct axis labels and units

Fixes #31482. 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/Muon/inc/MantidMuon/PlotAsymmetryByLogValue.h [hunk 1]
```diff
@@ -88,7 +88,9 @@ class MANTID_MUON_DLL PlotAsymmetryByLogValue : public API::Algorithm {
   /// Get log value
   double getLogValue(API::MatrixWorkspace &ws);
   /// Populate output workspace with results
-  void populateOutputWorkspace(API::MatrixWorkspace_sptr &outWS, int nplots);
+  void populateOutputWorkspace(API::MatrixWorkspace_sptr &outWS, int nplots, const std::string units);
+  /// get log units
+  std::string getLogUnits(const std::string fileName);
```
Comment (Matt-Cumber) [inline Framework/Muon/inc/MantidMuon/PlotAsymmetryByLogValue.h:93]:
```suggestion
  const std::string getLogUnits(const std::string &fileName);
```

File: Framework/Muon/src/PlotAsymmetryByLogValue.cpp [hunk 2]
```diff
@@ -206,6 +208,22 @@ void PlotAsymmetryByLogValue::exec() {
   saveResultsToADS(outWS, nplots + 1);
 }
 
+std::string PlotAsymmetryByLogValue::getLogUnits(const std::string fileName) {
+  Workspace_sptr loadedWs = doLoad(fileName);
+  MatrixWorkspace_sptr ws;
+  // Check if workspace is a workspace group
+  WorkspaceGroup_sptr group = std::dynamic_pointer_cast<WorkspaceGroup>(loadedWs);
```
Comment (Matt-Cumber) [inline Framework/Muon/src/PlotAsymmetryByLogValue.cpp:215]: We could shorten this a bit to something like
```
// Check if workspace is a workspace group, if it is not, we only have 'red' data
if (WorkspaceGroup_sptr group = std::dynamic_pointer_cast<WorkspaceGroup>(loadedWs)) {
  ws = std::dynamic_pointer_cast<MatrixWorkspace>(group->getItem(m_red - 1));
} else {
  ws = std::dynamic_pointer_cast<MatrixWorkspace>(loadedWs);
}
```
But it is your choice which is more readable for maintainability

File: Framework/Muon/src/PlotAsymmetryByLogValue.cpp [hunk 3]
```diff
@@ -196,8 +196,10 @@ void PlotAsymmetryByLogValue::exec() {
   MatrixWorkspace_sptr outWS = create<Workspace2D>(nplots,         //  the number of plots
                                                    Points(npoints) //  the number of data points on a plot
   );
+  std::string units = getLogUnits(m_fileNames[0]);
```
Comment (Matt-Cumber) [inline Framework/Muon/src/PlotAsymmetryByLogValue.cpp:199]:
```suggestion
  const auto units = getLogUnits(m_fileNames[0]);
```

File: Framework/Muon/src/PlotAsymmetryByLogValue.cpp [hunk 4]
```diff
@@ -206,6 +208,22 @@ void PlotAsymmetryByLogValue::exec() {
   saveResultsToADS(outWS, nplots + 1);
 }
 
+std::string PlotAsymmetryByLogValue::getLogUnits(const std::string fileName) {
```
Comment (Matt-Cumber) [inline Framework/Muon/src/PlotAsymmetryByLogValue.cpp:211]:
```suggestion
const std::string PlotAsymmetryByLogValue::getLogUnits(const std::string &fileName) {
```

File: Framework/Muon/inc/MantidMuon/PlotAsymmetryByLogValue.h [hunk 5]
```diff
@@ -88,7 +88,9 @@ class MANTID_MUON_DLL PlotAsymmetryByLogValue : public API::Algorithm {
   /// Get log value
   double getLogValue(API::MatrixWorkspace &ws);
   /// Populate output workspace with results
-  void populateOutputWorkspace(API::MatrixWorkspace_sptr &outWS, int nplots);
+  void populateOutputWorkspace(API::MatrixWorkspace_sptr &outWS, int nplots, const std::string units);
```
Comment (Matt-Cumber) [inline Framework/Muon/inc/MantidMuon/PlotAsymmetryByLogValue.h:91]:
```suggestion
  void populateOutputWorkspace(API::MatrixWorkspace_sptr &outWS, int nplots, const std::string &units);
```

File: Framework/Muon/src/PlotAsymmetryByLogValue.cpp [hunk 6]
```diff
@@ -431,7 +449,8 @@ Workspace_sptr PlotAsymmetryByLogValue::loadCorrectionsFromFile(const std::strin
  *   @param outWS :: [input/output] Output workspace to populate
  *   @param nplots :: [input] Number of histograms
  */
-void PlotAsymmetryByLogValue::populateOutputWorkspace(MatrixWorkspace_sptr &outWS, int nplots) {
+void PlotAsymmetryByLogValue::populateOutputWorkspace(MatrixWorkspace_sptr &outWS, int nplots,
+                                                      const std::string units) {
```
Comment (Matt-Cumber) [inline Framework/Muon/src/PlotAsymmetryByLogValue.cpp:453]:
```suggestion
                                                      const std::string &units) {
```

**Review by Matt-Cumber — CHANGES_REQUESTED** [review]
The changes work as expected and it is good we now have units in the ALC plot. I have suggested some code style changes, it is better to pass strings by const reference (I may have missed some instances so you will need to double check). The rest is your preference.

File: Framework/Muon/src/PlotAsymmetryByLogValue.cpp (hunk 2 above)
Comment (AnthonyLim23) [reply Framework/Muon/src/PlotAsymmetryByLogValue.cpp:215]: I think we should leave it to be consistent with the rest of the code.

**Review by Matt-Cumber — APPROVED** [review]
Code looks good and works as described. There are 288 cppcheck errors which is as expected, so nothing new introduced here. All jobs were passing before

**Review by SilkeSchomann — APPROVED** [review]

**Comment by AnthonyLim23** [comment]
@mantidproject/gatekeepers could we please get this merged? The build seems to have got stuck from the other day (but did pass previously)

**Comment by DannyHindson** [comment]
Static analysis job got reset on this and a few other PRs last week during a GitHub outage. Not sure why but merging despite the "Pending" status of this job because it had previously passed all checks

**Pull request merged**